presto-main/src/main/java/com/facebook/presto/server/HttpRemoteTask.java

```diff
@@ -380,6 +395,15 @@ private synchronized void updateTaskInfo(TaskInfo newValue, List<TaskSource> sou
                 pendingSourceSplitCount -= removed;
             }
         }
+
+        fireSplitChanged(getPartitionedSplitCount() - oldSplitCount);
+    }
+
+    private synchronized void fireSplitChanged(int delta)
+    {
+        if (splitCountChangeListener != null && delta != 0) {
+            splitCountChangeListener.splitCountChanged(delta);
```

Comment:
I think it's generally best to avoid making outcalls in a synchronized method, and I think you can make `splitCountChangeListener` final if you delete the `removeSplitCountChangeListener` method, unless I missed something